### PR TITLE
Convert `profiling`, `uber-plugins` and `versioning` buildSrc subprojects

### PR DESCRIPTION
**Pull request by lacasseio**
### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Fgradlebuild%2Flazy-conversion)
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt [hunk 1]
```diff
@@ -27,11 +28,11 @@ import java.net.URL
...
     override fun apply(project: Project): Unit = project.run {
-        wrapperUpdateTask("nightly", "nightly")
-        wrapperUpdateTask("rc", "release-candidate")
-        wrapperUpdateTask("current", "current")
+        wrapperUpdateTask(project.getTasks(), "nightly", "nightly")
+        wrapperUpdateTask(project.getTasks(), "rc", "release-candidate")
+        wrapperUpdateTask(project.getTasks(), "current", "current")
 
-        tasks.withType<Wrapper>() {
+        tasks.withType(Wrapper::class.java).configureEach {
```
[the remark this answers is not in the document]
Comment (big-guy) [reply buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt]: Yeah, I think we can do this:

`tasks.withType<Wrapper>().configureEach {`

[the remark this answers is not in the document]
Comment (lacasseio) [reply buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt]: Must have been the `apply { plugin("org.gradle.kotlin.kotlin-dsl") }` confusion that led me to believe this didn't work. Thanks, I will change that.

File: buildSrc/subprojects/versioning/versioning.gradle.kts [hunk 2]
```diff
@@ -2,6 +2,8 @@ plugins {
     `java-gradle-plugin`
 }
 
+apply { plugin("org.gradle.kotlin.kotlin-dsl") }
```
[the remark this answers is not in the document]
Comment (lacasseio) [reply buildSrc/subprojects/versioning/versioning.gradle.kts]: Thanks, I will change it and keep in mind for the other code that I will change.

File: buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt [hunk 3]
```diff
@@ -44,22 +45,22 @@ class WrapperPlugin : Plugin<Project> {
     }
 
     private
-    fun Project.wrapperUpdateTask(name: String, label: String) {
+    fun Project.wrapperUpdateTask(tasks: TaskContainer, name: String, label: String) {
```
[the remark this answers is not in the document]
Comment (lacasseio) [reply buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt]: Good point, I'm still confused a bit with the `Project` extension but I will get the hang of it.

File: buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt [hunk 4]
```diff
@@ -27,11 +28,11 @@ import java.net.URL
 class WrapperPlugin : Plugin<Project> {
 
     override fun apply(project: Project): Unit = project.run {
-        wrapperUpdateTask("nightly", "nightly")
-        wrapperUpdateTask("rc", "release-candidate")
-        wrapperUpdateTask("current", "current")
+        wrapperUpdateTask(project.getTasks(), "nightly", "nightly")
```
[the remark this answers is not in the document]
Comment (lacasseio) [reply buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/WrapperPlugin.kt]: Good point, I will change.

**Comment by lacasseio** [comment]
I updated everything @bamboo suggested.

**Pull request merged**